substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java

```diff
@@ -36,6 +36,9 @@
  */
 
 public class Range {
+
+    private static final char CLASS_DELIMITER = '.';
```

Comment:
This change is fine but it is going to change the way both VS and ELF debug info is generated. In the latter case that will cause the mx command mx testdebuginfo and the gate test which drives it to fail. Fixing this requires changing the test script mx.substartevm/testhello.py that drives a gdb session to also expect this separator. Occurrences of :: in both the input commands and the regular expressions used to match debugger output must be modified to a '.' character. n.b. That requires escaping the '.' in the regular expressions and attention needs to be paid to the type of string (raw or normall) in order to get the right number of backslashes (yeah using '.' is a pain in this regard even though it is better from a Java user point of view). I'll do this offline and forward you the edited version to try once you have reworked the code to address the latest round of review feedback.